ZhuoHeiChaAPI/Services/GameService.cs

```diff
@@ -125,18 +170,21 @@ public PlayHandReturn PlayHand(int gameId, int playerId, List<Card> UserCard)
             }
         }
 
-        public IEnumerable<PlayerType> GetPlayerTypeList(int gameId)
+        public IEnumerable<bool> IsBlackAceList(int gameId)
         {
             // assume gameId is always valid since it's called after ReturnTribute succeeded
             if (!_gameSessions.TryGetValue(gameId, out var gameLockPair))
             {
                 throw new Exception($"Failed to get playerTypes for game {gameId}");
             }
 
+            // TODO
+            // ???????????? the get operation do not need lock? ??????????????
```

Comment:
Yes you are right. We don't need lock for reading the black ace list here since during the execution of this method, no other threads could come in and change its value. The black ace list is determined exactly once per game during `InitGame`, which will not interfere with this method.

Reply:
Done